### PR TITLE
Upgrade to django 1.7

### PR DESCRIPTION
**Pull request by txels**
Upgrade to django 1.7:
- Move south migrations to django migrations
- Replace django-registration with a maintained 1.7-ready version
- Fix things that did not work

Additionally I realised now flake8 was coming up with  a bunch of issues, so I fixed them.

I run `./manage.py migrate --fake` in order to swallow the new migrations without killing my data.

### CODE REVIEW

File: ployst/core/features/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,102 +1,32 @@
...
-    complete_apps = ['features']
\ No newline at end of file
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
```
Comment (alexcouper) [inline ployst/core/features/migrations/0001_initial.py:9]: This looks awesome

File: ployst/core/repos/models.py [hunk 2]
```diff
@@ -16,13 +15,11 @@ class Revision(models.CharField):
...
     def __init__(self, *args, **kwargs):
         kwargs['max_length'] = 40
-        super(Revision, self).__init__(self, *args, **kwargs)
+        super(Revision, self).__init__(*args, **kwargs)
 
     def __unicode__(self):
         "The display name will be the column name"
-        return self.column
-
-add_introspection_rules([], ["^ployst\.core\.repos\.models\.Revision"])
```
Comment (alexcouper) [inline ployst/core/repos/models.py:25]: Wow. I can't remember what this was for...

File: ployst/settings/test.py [hunk 3]
```diff
@@ -18,4 +18,15 @@
...
+# http://stackoverflow.com/a/25267435/166761
+
+class DisableMigrations(object):
+    def __contains__(self, item):
+        return True
+
+    def __getitem__(self, item):
+        return "notmigrations"
+
+MIGRATION_MODULES = DisableMigrations()
```
Comment (alexcouper) [inline ployst/settings/test.py:32]: Nice

**Comment by alexcouper** [comment]
Merging

File: ployst/core/repos/models.py (hunk 2 above)
Comment (txels) [inline ployst/core/repos/models.py:25]: South needs this for custom fields.

**Pull request merged**